optuna/integration/_lightgbm_tuner/sklearn.py

```diff
@@ -1,49 +1,854 @@
+            specified.
+
+        min_split_gain:
+            Minimum loss reduction required to make a further partition on a
+            leaf node of the tree.
+
+        min_child_weight:
+            Minimum sum of instance weight (hessian) needed in a child (leaf).
+
+        min_child_samples:
+            Minimum number of data needed in a child (leaf).
+
+        subsample:
+            Subsample ratio of the training instance.
+
+        subsample_freq:
+            Frequence of subsample. <=0 means no enable.
+
+        colsample_bytree:
+            Subsample ratio of columns when constructing each tree.
+
+        reg_alpha:
+            L1 regularization term on weights.
+
+        reg_lambda:
+            L2 regularization term on weights.
+
+        random_state:
+            Seed of the pseudo random number generator. If int, this is the
+            seed used by the random number generator. If
+            ``numpy.random.RandomState`` object, this is the random number
+            generator. If None, the global random state from ``numpy.random``
+            is used.
+
+        n_jobs:
+            Number of parallel jobs. -1 means using all processors.
+
+        silent:
+            If true, print messages while running boosting.
+
+        importance_type:
+            Type of feature importances. If 'split', result contains numbers of
+            times the feature is used in a model. If 'gain', result contains
+            total gains of splits which use the feature.
+
+        study:
+            Study corresponds to the optimization task. If None, a new study is
+            created.
+
+        timeout:
+            Time limit in seconds for the search of appropriate models. If
+            None, the study is executed without time limitation. If
+            ``n_trials`` is also set to None, the study continues to create
+            trials until it receives a termination signal such as Ctrl+C or
+            SIGTERM. This trades off runtime vs quality of the solution.
+
+        model_dir:
+            Directory for storing the files generated during training.
+
+        **kwargs:
+            Other parameters for the model. See
+            http:/lightgbm.readthedocs.io/en/latest/Parameters.html for more
+            parameters. Note, that **kwargs is not supported in sklearn, so it
+            may cause unexpected issues.
+
+    Attributes:
+        encoder_:
+            Label encoder.
+
+    Examples:
+        >>> import optuna.integration.lightgbm as lgb
+        >>> from sklearn.datasets import load_iris
+        >>> from sklearn.model_selection import train_test_split
+        >>> clf = lgb.LGBMClassifier(random_state=0)
+        >>> X, y = load_iris(return_X_y=True)
+        >>> X, X_valid, y, y_valid = train_test_split(X, y, random_state=0)
+        >>> clf.fit(X, y, eval_set=[(X_valid, y_valid)])
+        LGBMClassifier(...)
+        >>> y_pred = clf.predict(X)
```

Comment:
Please use the [`doctest` directive](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#directive-doctest) for a doc-test style code block. In addition, could you set `verbose=False`? Currently, this example produces so many log messages.

```suggestion
    Examples:

        .. doctest::

            >>> import optuna.integration.lightgbm as lgb
            >>> from sklearn.datasets import load_iris
            >>> from sklearn.model_selection import train_test_split
            >>> clf = lgb.LGBMClassifier(random_state=0)
            >>> X, y = load_iris(return_X_y=True)
            >>> X, X_valid, y, y_valid = train_test_split(X, y, random_state=0)
            >>> clf.fit(X, y, eval_set=[(X_valid, y_valid)], verbose=False)
            LGBMClassifier(...)
            >>> y_pred = clf.predict(X)
```

In addition, this example causes an error due to the unexpected warning message (i.e., ` [LightGBM] [Warning] min_data_in_leaf is set=20, min_child_samples=20 will be ignored. Current value: min_data_in_leaf=20` ). Do you have any idea to suppress it?

```console
Warning, treated as error:
**********************************************************************
File "../../optuna/integration/_lightgbm_tuner/sklearn.py", line ?, in default
Failed example:
    clf.fit(X, y, eval_set=[(X_valid, y_valid)], verbose=False)
Expected:
    LGBMClassifier(...)
Got:
    [LightGBM] [Warning] min_data_in_leaf is set=20, min_child_samples=20 will be ignored. Current value: min_data_in_leaf=20
    LGBMClassifier(boosting_type='gbdt', class_weight=None, colsample_bytree=1.0,
                   importance_type='split', learning_rate=0.1, max_depth=-1,
                   min_child_samples=20, min_child_weight=0.001, min_split_gain=0.0,
                   model_dir=None, n_estimators=100, n_jobs=-1, num_leaves=31,
                   objective=None, random_state=0, reg_alpha=0.0, reg_lambda=0.0,
                   silent=True, study=None, subsample=1.0, subsample_for_bin=200000,
                   subsample_freq=0, timeout=None)

make: *** [Makefile:25: doctest] Error 2
```

Reply:
Fixed in 9351d0f.

> In addition, this example causes an error due to the unexpected warning message (i.e., `[LightGBM] [Warning] min_data_in_leaf is set=20, min_child_samples=20 will be ignored. Current value: min_data_in_leaf=20` ). Do you have any idea to suppress it?

https://github.com/optuna/optuna/blob/master/optuna/integration/_lightgbm_tuner/alias.py#L9-L12
`min_data_in_leaf`, `min_data_per_leaf`, `min_data`, and `min_child_samples` will be renamed to `min_data_in_leaf` by `_handling_alias_parameters`.

https://github.com/optuna/optuna/blob/master/optuna/integration/_lightgbm_tuner/optimize.py#L594-L599
On the other hand, `LightGBMTuner` will try to tune `min_child_samples`.

In this case, `min_data_in_leaf` conflicts with `min_child_samples`,  and `min_chile_samples`, which should be tuned, will be ignored. I think we can solve this problem by renaming all `min_child_samples` that appear in `optimize.py` to `min_data_in_leaf`.